optimum/exporters/tasks.py

```diff
@@ -1421,6 +1444,100 @@ def infer_task_from_model(
 
         return task
 
+    @classmethod
+    def infer_library_from_model(
+        cls,
+        model_name_or_path: str,
+        subfolder: str = "",
+        revision: Optional[str] = None,
+    ):
+        """
+        Infers the library from the model repo.
+
+        Args:
+            model_name_or_path (`str`):
+                The model to infer the task from. This can either be the name of a repo on the HuggingFace Hub, an
+                instance of a model, or a model class.
+            subfolder (`str`, *optional*, defaults to `""`):
+                In case the model files are located inside a subfolder of the model directory / repo on the Hugging
+                Face Hub, you can specify the subfolder name here.
+            revision (`Optional[str]`, *optional*, defaults to `None`):
+                Revision is the specific model version to use. It can be a branch name, a tag name, or a commit id.
+
+        Returns:
+            `str`: The library name automatically detected from the model repo.
+        """
+        library_name = None
+
+        full_model_path = Path(model_name_or_path) / subfolder
+        if full_model_path.is_dir():
+            all_files = list(full_model_path.glob("**/*"))
+            if "model_index.json" in all_files:
+                library_name = "diffusers"
+            if "config.json" in all_files:
+                with open("config.json") as fp:
+                    cfg = json.load(fp)
+
+                if "pretrained_cfg" in cfg:
+                    library_name = "timm"
+                elif "_diffusers_version" in cfg:
+                    library_name = "diffusers"
+                else:
+                    library_name = "transformers"
+        else:
+            library_name = huggingface_hub.model_info(model_name_or_path, revision=revision).library_name
+
+        return library_name
+
+    @classmethod
+    def patch_model_for_export(
```

Comment:
Maybe we could find an other name in order not to confuse with the model patcher from the ONNX export. Something like `standardize_model_attributes` for example

Reply:
Done